controller/api/public/edges.go

```diff
@@ -62,8 +62,13 @@ func (s *grpcServer) getEdges(ctx context.Context, req *pb.EdgesRequest) ([]*pb.
 	}
 	resourceType := string(labelNames[1]) // skipping first name which is always namespace
 	labels := promQueryLabels(req.Selector.Resource)
-	labelsOutbound := labels.Merge(promDirectionLabels("outbound"))
-	labelsInbound := labels.Merge(promDirectionLabels("inbound"))
+	labelsOutbound := fmt.Sprint(labels.Merge(promDirectionLabels("outbound")))
+	labelsInbound := fmt.Sprint(labels.Merge(promDirectionLabels("inbound")))
+
+	// checking that data for the selected resource type exists
+	resourceExistsCheck := fmt.Sprintf(`%s!="",`, resourceType)
+	labelsOutbound = labelsOutbound[:1] + resourceExistsCheck + labelsOutbound[1:]
+	labelsInbound = labelsInbound[:1] + resourceExistsCheck + labelsInbound[1:]
```

Comment:
:sob: I don't know if this would be any less weird/confusing, but we could do what the prometheus library does with stringifying label sets, but also add in `!=` capabilities...

```
func CustomToString(l LabelSet, resourceType string) string {
    lstrs := make([]string, 0, len(l))
    for l, v := range l {
        lstrs = append(lstrs, fmt.Sprintf("%s=%q", l, v))
    }
  lstrs = append(lstrs, fmt.Sprintf(`%s!="",`, resourceType))

    sort.Strings(lstrs)
    return fmt.Sprintf("{%s}", strings.Join(lstrs, ", "))
}
```